### ISSUE TITLE
Add support for $removeparam modifier

### ISSUE THREAD

**Issue by slavaleleka** [ameshkov/VscodeAdblockSyntax#41]
<details>
  <summary>example 1</summary>

```
$removeparam=_openstat
$removeparam=action_object_map
$removeparam=action_ref_map
$removeparam=utm_content
```

  ![Screenshot 1](https://uploads.adguard.com/slbdzdrdw2ia.png)
</details>

<details>
  <summary>example 2</summary>

```
||app.mi.com/download/*?id=$removeparam=ref
||mp.weixin.qq.com^$removeparam=pass_ticket
||visit.digidip.net^$removeparam=/^(ppref|ref|pid)=/
||mvideo.ru^$removeparam=/^(_requestid|reff)=/
```

  ![Screenshot 2](https://uploads.adguard.com/slbdzdrtd21w.png)
</details>

**Comment by bogachenko** [comment]
Hi @ameshkov, is this a bug? The ADGfW program does not allow creating a regexp with repetitions. The `{2,4}` construction does not work for this modifier, everything after the comma causes an error. I can only create `{2}`, why?
eg. `$removeparam=/ref(err?al)?([a-zA-Z0-9]{2,3})?[_=]/` or `$removeparam=/test[1-9]{2,4}/`

**Comment by Alex-302** [comment]
Try with escaped comma - `{2\,4}`

**Comment by bogachenko** [comment]
Yes, it works. But why should this be written, because it completely breaks the regexp [markup](https://regex101.com/r/0XRnif/1).

**Comment by Alex-302** [comment]
https://adguard.com/kb/general/ad-filtering/create-own-filters/#domain-modifier - applied to all regexps in AG

**Comment by bogachenko** [comment]
That's not what I was talking about, I don't understand why you're answering the question that I didn't ask.

As far as I can tell, the ADG program was universal. She has her own sheets, but she tries to maintain the sheets and the Frenchy's achievements. There is a clear algorithm for regexps, it does not differ, and this is the standard when the number of repetitions and the "comma" symbol are wrapped in curly brackets, there can be no extraneous characters in it and the link that you gave does not apply to my question, since it says about the non-wrapped "comma" sign.

In other words, as I said, edits have been made to the markup of the regex for the program, it differs from the standard, but there is no word about this in the wiki, plus support from the Frenchy sheets will be broken.

**Comment by scripthunter7** [comment]
@bogachenko This is not a bug, it is simply a limitation of the current syntax (dollar sign and comma are special characters in adblock syntax). There are plans in the near future to eliminate the need for extra escapes in regexps.

Reference: https://github.com/AdguardTeam/tsurlfilter/issues/121

**Comment by slavaleleka** [comment]
> It's also not on your [wiki](https://adguard.com/kb/general/ad-filtering/create-own-filters/).

special character escaping for `$removeparam` is actually mentioned in the kb:
https://uploads.adguard.org/slGBfPXuJS31USt5Ez.png
https://adguard.com/kb/general/ad-filtering/create-own-filters/#removeparam-modifier

**Comment by bogachenko** [comment]
> > It's also not on your [wiki](https://adguard.com/kb/general/ad-filtering/create-own-filters/).
> 
> special character escaping for `$removeparam` is actually mentioned in the kb: https://uploads.adguard.org/slGBfPXuJS31USt5Ez.png https://adguard.com/kb/general/ad-filtering/create-own-filters/#removeparam-modifier

I was talking about a wrapped special sign, not a regular one. Since I started from the standard regexp markup, of course I know that the usual ones are excluded by the backslash character.
